src/inotify.rs

```diff
@@ -269,46 +266,9 @@ impl Inotify {
     /// initialized in `Inotify::init`. This is intended to be used to transform an
     /// `EventStream` back into an `Inotify`. Do not attempt to clone `Inotify` with this.
     #[cfg(feature = "stream")]
-    pub(crate) fn from_file_descriptor(fd: Arc<FdGuard>) -> Self {
+    pub(crate) fn from_file_descriptor(fd: Arc<OwnedFd>) -> Self {
         Inotify { fd }
     }
-
-    /// Closes the inotify instance
-    ///
-    /// Closes the file descriptor referring to the inotify instance. The user
-    /// usually doesn't have to call this function, as the underlying inotify
-    /// instance is closed automatically, when [`Inotify`] is dropped.
-    ///
-    /// # Errors
-    ///
-    /// Directly returns the error from the call to [`close`], without adding any
-    /// error conditions of its own.
-    ///
-    /// # Examples
-    ///
-    /// ```
-    /// use inotify::Inotify;
-    ///
-    /// let mut inotify = Inotify::init()
-    ///     .expect("Failed to initialize an inotify instance");
-    ///
-    /// inotify.close()
-    ///     .expect("Failed to close inotify instance");
-    /// ```
-    ///
-    /// [`close`]: libc::close
-    pub fn close(self) -> io::Result<()> {
-        // `self` will be dropped when this method returns. If this is the only
-        // owner of `fd`, the `Arc` will also be dropped. The `Drop`
-        // implementation for `FdGuard` will attempt to close the file descriptor
-        // again, unless this flag here is cleared.
-        self.fd.should_not_close();
-
-        match unsafe { ffi::close(**self.fd) } {
-            0 => Ok(()),
-            _ => Err(io::Error::last_os_error()),
-        }
-    }
```

Comment:
I'd rather keep this method as a no-op. The documentation should change accordingly.

I'd be fine with deprecating it, as it's superfluous. But I'd like to keep the breaking changes to a minimum